core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/RowDenormalizerAggregator.java

```diff
@@ -139,36 +138,39 @@ private Schema initializeOutputSchema() {
 
   /**
    * @param inputSchema Validates whether the keyfield, fieldname, and fieldvalue entered by the user is of type
-   *                    String and present in the input schema or not.
+   *                    String or Nullable String and present in the input schema or not.
    * @return true
    */
   private void validateInputFields(Schema inputSchema) {
 
     Schema.Field keyField = inputSchema.getField(conf.getKeyField());
-    Schema.Field fieldName = inputSchema.getField(conf.getFieldName());
-    Schema.Field fieldValue = inputSchema.getField(conf.getFieldValue());
+    Schema.Field nameField = inputSchema.getField(conf.getNameField());
+    Schema.Field valueField = inputSchema.getField(conf.getValueField());
 
     if (keyField == null) {
       throw new IllegalArgumentException(
         String.format("Keyfield '%s' does not exist in input schema %s", conf.getKeyField(), inputSchema));
-    } else if (fieldName == null) {
+    } else if (nameField == null) {
       throw new IllegalArgumentException(
-        String.format("Fieldname '%s' does not exist in input schema %s", conf.getFieldName(), inputSchema));
-    } else if (fieldValue == null) {
+        String.format("Namefield '%s' does not exist in input schema %s", conf.getNameField(), inputSchema));
+    } else if (valueField == null) {
       throw new IllegalArgumentException(
-        String.format("Fieldvalue '%s' does not exist in input schema %s", conf.getFieldValue(), inputSchema));
+        String.format("Valuefield '%s' does not exist in input schema %s", conf.getValueField(), inputSchema));
     } else if (!((keyField.getSchema().isNullable() ? keyField.getSchema().getNonNullable().getType() : keyField
```

Comment:
this is really hard to read. Can you create a `Schema.Type keyFieldType` variable instead of doing all this in an if clause? You can also directly compare enums with '=='.